google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/HttpApi.scala

```diff
@@ -128,9 +128,13 @@ private trait HttpApi {
     for {
       request <- Marshal((HttpMethods.POST, url, request)).to[HttpRequest]
       response <- doRequest(request, maybeAccessToken)
-      publishResponse <- Unmarshal(response.entity).to[PublishResponse]
-    } yield publishResponse.messageIds
-  }
+    } yield {
+      response.status match {
+        case StatusCodes.OK => Unmarshal(response.entity).to[PublishResponse].map(_.messageIds)
+        case _ => throw new RuntimeException(s"Unexpected publish response. Code: [${response.status}]")
```

Comment:
Does the response contain anything else that might be useful? An error message possibly?
It might be good to introduce an exception class for this instead of using RTE.

Reply:
Unfortunately it doesn't contain any more info. Just 404.

Reply:
Yes, custom exception might be good. Let's improve it later.